gui/gui.go

```diff
@@ -179,22 +184,57 @@ func (ui *GUI) loadTemplates() error {
 
 // Run starts the user interface.
 func (ui *GUI) Run() {
-	log.Infof("GUI served on port %v.", ui.cfg.GUIPort)
-
 	go func() {
-		if err := ui.server.ListenAndServeTLS(ui.cfg.TLSCertFile,
-			ui.cfg.TLSKeyFile); err != nil &&
-			err != http.ErrServerClosed {
-			log.Error(err)
+		if !ui.cfg.UseLEHTTPS {
+			ui.server = &http.Server{
+				WriteTimeout: time.Second * 30,
+				ReadTimeout:  time.Second * 30,
+				IdleTimeout:  time.Second * 30,
+				Addr:         fmt.Sprintf("0.0.0.0:%v", ui.cfg.GUIPort),
+				Handler:      ui.router,
+			}
+
+			if err := ui.server.ListenAndServeTLS(ui.cfg.TLSCertFile,
+				ui.cfg.TLSKeyFile); err != nil &&
+				err != http.ErrServerClosed {
+				log.Error(err)
+			}
 		}
-	}()
-}
 
-// Shutdown tears down the user interface.
-func (ui *GUI) Shutdown() {
-	ctx, cl := context.WithTimeout(ui.cfg.Ctx, time.Second*5)
-	defer cl()
-	if err := ui.server.Shutdown(ctx); err != nil {
-		log.Error(err)
-	}
+		if ui.cfg.UseLEHTTPS {
```

Comment:
Could use `else` here rather than checking `UseLEHTTPS==true` and then checking `UseLEHTTPS==false`?

It would help make it really clear that we are only want to run one code block or the other, and never both.

Reply:
For long code blocks if-else isn't ideal because "else" doesn't provide any detail on the condition being checked. You have to scroll back and infer. I think separating the code blocks, explicitly showing the conditions being checked reads better.

Reply:
The problem with else when having a verification with AND is because according to De Morgan law: 
 `~(P && Q) == (~P || ~Q)`

And it can end having complication and unexpected results. So I agree with @dnldd on this one.

Reply:
Im not sure what you are referring to vctt94. These if conditions have no `OR`s or `AND`s.

Fair enough, lets leave it in, I just wanted to double check it wasnt an oversight

Reply:
sorry. I thought it was about https://github.com/decred/dcrpool/pull/83/files#diff-b4bda758a2aef091432646c354b4dc59R481